itou/openid_connect/models.py

```diff
@@ -98,7 +99,7 @@ class OIDConnectUserData:
     username: str
     identity_provider: IdentityProvider
     kind: str
-    login_allowed_user_kinds = [UserKind]
+    login_allowed_user_kinds: ClassVar[list[UserKind]]
```

Comment:
Si `login_allowed_user_kinds` est une `ClassVar` alors `identity_provider` et `kind` le sont très probablement aussi ;).

Reply:
C'est la seule façon que j'ai trouvé pour réussi à filer `list[UserKind]` :  
Sans ça done on obtient une `ValueError: mutable default <class 'list'> for field login_allowed_user_kinds is not allowed: use default_factory`

Reply:
Oui c'est ce que j'ai vu, au début je me suis dit qu'on avais pas spécifiquement besoin d'une `list` et qu'on pourrais avoir une structure immutable mais le `ClassVar` est tout aussi vrai vu qu'on veux la même valeur pour toutes les instances de classe, ce qui m'a fait penser que les 2 autres devraient aussi en être.
En soi c'est pas bloquant mais c'est plus pour éviter les Oo quand on arrive dessus et que un seul l'est alors que 3 sont utilisés de la même manière.

Reply:
Ah, je comprends mieux.
On ne peut pas le mettre sur les 2 autres parce que sinon, on ne les récupère pas dans `user_data_dict = dataclasses.asdict(self)` et on il manque des champs pour créer l'utilisateur

Reply:
Ah bah oui :facepalm: :+1:

Reply:
Pour info : j'ai extrait cette partie pour la mettre dans cette PR : https://github.com/gip-inclusion/les-emplois/pull/4822